components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java

```diff
@@ -872,4 +875,36 @@ private APIError handleIdentityUserStoreException(IdentityUserStoreMgtException
         errorResponse.setDescription(exception.getMessage());
         return new APIError(status, errorResponse);
     }
+
+    /**
+     * Validate userstore update request.
+     *
+     * @param domainID     Userstore domain ID
+     * @param userStoreReq UserStoreReq object.
+     * @throws IdentityUserStoreClientException If request is invalid.
+     */
+    private void validateUserstoreUpdateRequest(String domainID, UserStoreReq userStoreReq)
+            throws IdentityUserStoreClientException {
+
+        if (StringUtils.isBlank(domainID)) {
+            throw new IdentityUserStoreClientException(
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_EMPTY_DOMAIN_ID.getCode(),
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_EMPTY_DOMAIN_ID.getMessage());
+        }
+        if (userStoreReq == null) {
+            throw new IdentityUserStoreClientException(
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_REQUEST_BODY_NOT_FOUND.getCode(),
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_REQUEST_BODY_NOT_FOUND.getMessage());
+        }
+        if (StringUtils.isBlank(userStoreReq.getName())) {
+            throw new IdentityUserStoreClientException(
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_EMPTY_DOMAIN_NAME.getCode(),
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_EMPTY_DOMAIN_NAME.getMessage());
+        }
+        if (!userStoreReq.getName().equals(base64URLDecodeId(domainID))) {
+            throw new IdentityUserStoreClientException(
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_EMPTY_DOMAIN_NAME.getCode(),
```

Comment:
is this error code correct?

Reply:
Fixed 0769df00c8e498d9efdbb1f8827130f28567c8de